### PR TITLE
feat(geometry): Rotate GeoJSON Polygons when a right angle is found

### PR DESCRIPTION
**Pull request by chriswmackey**
This commit adds new methods that rotate the GeoJSON polygon if it finds that the lower left corner of the polygon has a right angle. This way, GHEDesigner will place the grid of boreholes along the site boundaries, which is more spatially efficient and closer to what people would actually build.

The commit also refactors some of the methods that do pure geometry manipulation into a `geometry.py` module that is separate from the `projection.py` module that converts between lat/long and meters.

To illustrate the change that this commit makes, it effectively turns the following borehole layout that the current package produces:

image

.. into this:

image

### CODE REVIEW

**Comment by mitchute** [comment]
@vtnate do you know why are tests not being run on this? Based on the [actions repo settings](https://github.com/NREL/ThermalNetwork/settings/actions), it seems to me like they should run automatically. Or that we should get a prompt to approve tests, and then they should run.

File: thermalnetwork/projection.py
Comment (mitchute) [inline thermalnetwork/projection.py:3]: I'd prefer the full namespace here.

**Review by mitchute — COMMENTED** [review]
Thanks @chriswmackey. Looks good. One of the test results changed, but I'm guessing that's not due to a problem and is just due to the field rotation. Let me know if you think that's right. If so, I think you can just update the test value and we can merge this.

**Comment by mitchute** [comment]
Here's the results from the failing test.

```python
self = <test_network.TestNetwork testMethod=test_network_one_ghe_detailed>

    def test_network_one_ghe_detailed(self):
        # -- Set up
        output_path = self.test_outputs_path / "one_ghe_detailed_geo"
        output_path.mkdir(parents=True, exist_ok=True)
    
        # -- Run
        run_sizer_from_cli_worker(
            self.system_parameter_path_1_ghe_geometry,
            self.scenario_directory_path_1_ghe,
            self.geojson_file_path_1_ghe,
            output_path,
        )
    
        expected_outputs = {"8c369df2-18e9-439a-8c25-875851c5aaf0": {"num_bh": 82, "depth": 133.5}}
    
>       self.check_outputs(output_path, expected_outputs)

thermalnetwork/tests/test_network.py:39: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
thermalnetwork/tests/test_base.py:92: in check_outputs
    self.assertEqual(actual_nbh, expected_nbh)
E   AssertionError: 77 != 82
------------------------------------
```

**Comment by vtnate** [comment]
@chriswmackey This is a great improvement! Those pictures are valuable.

@mitchute Perhaps because this PR is coming from a fork and not our own repo? CI is set to run on every push. I will investigate the github settings and update our CI yaml file.

**Comment by chriswmackey** [comment]
Thanks, @mitchute . Sorry that I forgot to update the unit test. I just pushed an update for it and all of them now pass on my local machine:

image

I can also change this to be a PR from a branch instead of my fork if that's helpful but it looks like you already have a fix for that [here](https://github.com/NREL/ThermalNetwork/pull/53), @vtnate .

**Comment by vtnate** [comment]
@chriswmackey, can you run `pre-commit run --all` on your branch to run the linter/formatter?

**Comment by chriswmackey** [comment]
Thanks, @vtnate . I ran the linter and committed the auto-updated files. Hopefully, this gets all of the tests to pass.

**Comment by vtnate** [comment]
The code looks good to me, though I think @mitchute should review/approve for scientific appropriateness before we merge it.

**Comment by mitchute** [comment]
I'm good with it. Feel free to merge this.

**Review by vtnate — APPROVED** [review]

**Pull request merged**
